conans/test/functional/cross_building/build_requires_from_profile_test.py

```diff
@@ -52,6 +52,5 @@ def test_br_from_profile_host_and_profile_build(self):
         t.run("create library.py --profile:host=profile_host --profile:build=profile_build --build *")
         self.assertNotIn("br1/version: Applying build-requirement: br2/version", t.out)
         self.assertIn("br1/version: Applying build-requirement: br3/version", t.out)
-        self.assertIn("br2/version: Applying build-requirement: br3/version", t.out)
```

Comment:
Of course this one fails, but we need to write down if this is a known bug we don't want to fix right now or it is a design decision not to apply BRs to BRs when they are defined in profiles.